circuits/cpp/barretenberg/cpp/src/barretenberg/proof_system/relations/ecc_vm/ecc_transcript_relation.cpp

```diff
@@ -0,0 +1,263 @@
+     * this!
+     */
+    auto add_msm_into_accumulator = msm_transition * (-is_accumulator_empty + 1);
+    auto x3 = transcript_accumulator_x_shift;
+    auto y3 = transcript_accumulator_y_shift;
+    auto x1 = transcript_accumulator_x;
+    auto y1 = transcript_accumulator_y;
+    auto x2 = transcript_msm_x;
+    auto y2 = transcript_msm_y;
+    auto tmpx = (x3 + x2 + x1) * (x2 - x1) * (x2 - x1) - (y2 - y1) * (y2 - y1);
+    auto tmpy = (y3 + y1) * (x2 - x1) - (y2 - y1) * (x1 - x3);
+    std::get<7>(accumulator) += tmpx * add_msm_into_accumulator * scaling_factor; // degree 5
+    std::get<8>(accumulator) += tmpy * add_msm_into_accumulator * scaling_factor; // degree 4
+
+    /**
+     * @brief If is_accumulator_empty == 1, assign transcript_accumulator output into accumulator
+     *
+     * @note The accumulator point for all operations at row `i` is the accumulator point at row `i + 1`!
+     */
+    auto assign_msm_into_accumulator = msm_transition * is_accumulator_empty;
+    std::get<9>(accumulator) += assign_msm_into_accumulator * (x3 - x2) * scaling_factor; // degree 3
+    std::get<10>(accumulator) += assign_msm_into_accumulator * (y3 - y2) * scaling_factor;
+
+    /**
+     * @brief Constrain `add` opcode.
+     *
+     * add will add the input point in (transcript_x, transcript_y) into the accumulator.
+     * Correctly handles case where accumulator is point at infinity.
+     * TODO: need to add constraints to rule out point doubling case (x2 != x1)
+     * TODO: need to assert input point is on the curve!
+     */
+    x2 = transcript_x;
+    y2 = transcript_y;
+    auto add_into_accumulator = q_add * (-is_accumulator_empty + 1);
+    tmpx = (x3 + x2 + x1) * (x2 - x1) * (x2 - x1) - (y2 - y1) * (y2 - y1);
+    tmpy = (y3 + y1) * (x2 - x1) - (y2 - y1) * (x1 - x3);
+    std::get<11>(accumulator) += tmpx * add_into_accumulator * scaling_factor; // degree 5
+    std::get<12>(accumulator) += tmpy * add_into_accumulator * scaling_factor; // degree 4
+    auto assign_into_accumulator = q_add * is_accumulator_empty;
+    std::get<13>(accumulator) += (x3 - x2) * assign_into_accumulator * scaling_factor; // degree 3
+    std::get<14>(accumulator) += (y3 - y2) * assign_into_accumulator * scaling_factor;
+
+    /**
+     * @brief Opcode exclusion tests. We have the following assertions:
+     * 1. If q_mul = 1, (q_add, eq, reset) are zero
+     * 2. If q_reset = 1, is_accumulator_empty at next row = 1
+     * 3. If q_add = 1 OR msm_transition = 1, is_accumulator_empty at next row = 0
+     * 4. If q_add = 0 AND msm_transition = 0 AND q_reset_accumulator = 0, is_accumulator at next row = current row
+     * value
+     * @note point 3: both q_add = 1, msm_transition = 1 cannot occur because of point 1 (msm_transition only 1 when
+     * q_mul 1) we can use a slightly more efficient relation than a pure binary OR
+     */
+    std::get<15>(accumulator) += q_mul * (q_add + q_eq + q_reset_accumulator) * scaling_factor;
```

Comment:
The values of q_mal, q_add, q_eq, q_reset_accumulator need to be constrained to be 0 or 1